preprocessors/object-detection-azure/objdetect.py

```diff
@@ -0,0 +1,190 @@
+
+import requests  # pip3 install requests
+from re import search
+import operator
+
+# import numpy as np
+import json
+import time
+import jsonschema
+import logging
+import base64
+import os
+from flask import Flask, request, jsonify
+
+app = Flask(__name__)
+
+
+def process_results(result):
+    send = []
+    image_height = result["metadata"]["height"]
+    image_width = result["metadata"]["width"]
+    for item in range(len(result['objects'])):
+        dims = result["objects"][item]['rectangle']
+        x_left = dims['x'] / image_width
+        x_right = (dims['x'] + dims['w']) / image_width
+        y_left = dims['y'] / image_height
+        y_right = (dims['y'] + dims['h']) / image_height
+        centre = [abs((x_left + x_right) / 2),
+                  abs((y_left + y_right) / 2)]
+        area = abs((dims['w'] * dims['h']) / (image_height * image_width))
+        class_prob = float(result['objects'][item]['confidence'])
+        normalised_dims = [x_left, y_left, x_right, y_right]
+        dictionary = {"ID": item,
+                      "type": result['objects'][item]['object'],
+                      "dimensions": normalised_dims,
+                      "confidence": class_prob,
+                      "centroid": centre,
+                      "area": area
+                      }
+        send.append(dictionary)
+    return send
+
+
+def process_image(image):
+
+    region = "canadacentral"  # For example, "westus"
+    api_key = os.environ["AZURE_API_KEY"]
+
+    # Set request headers
+    headers = dict()
+    headers['Ocp-Apim-Subscription-Key'] = api_key
+    headers['Content-Type'] = 'application/octet-stream'
+
+    # Set request querystring parameters
+    """Only query categories"""
+    params = {'visualFeatures': 'Objects'}
+
+    # Make request and process response
+    response = requests.request(
+        'post',
+        "https://{}.api.cognitive.microsoft.com/vision/v1.0/analyze".format(region),
+        data=image,
+        headers=headers,
+        params=params
+    )
+    if response.status_code == 200 or response.status_code == 201:
+
+        if 'content-length' in response.headers and \
+                int(response.headers['content-length']) == 0:
+            return "", 204
+        elif 'content-type' in response.headers and \
+                isinstance(response.headers['content-type'], str):
+            if 'application/json' in response.headers['content-type'].lower():
+                if response.content:
+                    result = response.json()
+                    label = process_results(result)
+                else:
+                    return "", 204
+            else:
+                return "", 204
+
+    else:
+        return "", 204
+
+    return label
+
+
+
+@app.route("/preprocessor", methods=['POST', ])
+def categorise():
+    logging.debug("Received request")
+    # load the schema
+    with open('./schemas/preprocessors/object-detection.schema.json') \
```

Comment:
Why are schemas being loaded on each request? These files won't change and the resolver can just be reused.